### ISSUE TITLE
Lib size almost does not change when doing custom build on iOS compared to Android

### ISSUE THREAD

**Issue by hovhannesgithub** [pytorch/pytorch#35306]
## 🐛 Bug

On master branch when doing custom build for Android I get 40% decrease in Lib size (for libpytorch_jni.so) but when doing custom build for iOS with the same .yaml file I get about 2.7% decrease (for libtorch_cpu.a).

## To Reproduce

1. BUILD_PYTORCH_MOBILE=1 IOS_ARCH=arm64 ./scripts/build_ios.sh
2. SELECTED_OP_LIST=ops.yaml BUILD_PYTORCH_MOBILE=1 IOS_ARCH=arm64 ./scripts/build_ios.sh
3. ./scripts/build_pytorch_android.sh arm64-v8a
4. SELECTED_OP_LIST=ops.yaml ./scripts/build_pytorch_android.sh arm64-v8a

## Expected behavior

I think that on both platforms we should expect similar results. In this case 40% decrease of Lib size for iOS.

## Environment
 - PyTorch Version: master:
 - OS: MacOS Catalina 10.15.3:
 - cmake: 3.15.4:
 - gradle: 4.10.3:
 - Android NDK: 21.0.6113669:
 - Xcode: 11.3.1:

## Additional context

Also attaching all build outputs, yaml file that I have used and test log on how I get lib sizes.

[pytorch_android_build.log](https://github.com/pytorch/pytorch/files/4376885/pytorch_android_build.log)
[pytorch_android_custom_build.log](https://github.com/pytorch/pytorch/files/4376887/pytorch_android_custom_build.log)

[pytorch_ios_build.log](https://github.com/pytorch/pytorch/files/4376888/pytorch_ios_build.log)
[pytorch_ios_custom_build.log](https://github.com/pytorch/pytorch/files/4376894/pytorch_ios_custom_build.log)

[pytorch-test.log](https://github.com/pytorch/pytorch/files/4376897/pytorch-test.log)

[ops.yaml.zip](https://github.com/pytorch/pytorch/files/4376899/ops.yaml.zip)

**Comment by ArmanChopikyan** [comment]
I have a exact same problem.

## Environment
 - PyTorch Version: master:
 - OS: MacOS Catalina 10.15.3:
 - Xcode: 11.3.1:

## Additional context
Also add yaml file content.
`- aten::relu`

**Comment by xta0** [comment]
@ArmanChopikyan @hovhannesgithub The size reduction happens when you link your static libraries to output the final binary, as linker can strip most of the unused code (`-dead_strip`). Unlike dynamic libraries(.so in Android), linker can't strip code when compiling static libraries.

**Comment by hovhannesgithub** [comment]
@xta0 Thanks for info. Besides comparing .a file sizes I also checked size as described here:https://stackoverflow.com/questions/32003262/find-size-contributed-by-each-external-library-on-ios/39593377
Both ways gave insignificant difference. I tried also with -dead_strip but seams XCode is already doing it.
After that I tried simply comparing final .ipa files and result was as expected, about 40% decrease in size.
So closing this issue.
